app/scripts/model/dataProxy.js

```diff
@@ -1367,102 +1373,95 @@ window.DataManagerForIviz = (function($, _) {
-              _map[_panelMeta.stableId] = {};
-              var _sorted = (_panelMeta.samples).sort();
-              _map[_panelMeta.stableId].samples = _sorted;
-              _map[_panelMeta.stableId].sel_samples = _sorted;
-            });
-            fetch_promise.resolve(_map);
-          }).fail(function() {
-            // Silently fail gene panel sample id calls
-            fetch_promise.resolve({});
+          var _panels = _.filter(Object.keys(panelSamplesMap), function(item) {
+            return item !== wholeExomeSequencedPanelId;
           });
-        }
-      ),
-      getGenePanelMap: window.cbio.util.makeCachedPromiseFunction(
-        function(self, fetch_promise) {
-          self.getAllGenePanelSampleIds().then(function(_panelSampleMap) {
-            self.panelSampleMap = _panelSampleMap;
-            var asyncAjaxCalls = [];
-            var responses = [];
-            _.each(Object.keys(_panelSampleMap), function(_panelId) {
-              asyncAjaxCalls.push(
-                $.ajax({
-                  url: window.cbioURL + 'api-legacy/genepanel',
-                  contentType: 'application/json',
-                  data: {panel_id: _panelId},
-                  type: 'GET',
-                  success: function(_res) {
-                    responses.push(_res);
-                  }
-                })
-              );
-            });
-            $.when.apply($, asyncAjaxCalls).done(function() {
-              var _panelMetaArr = _.map(responses, function(responseArr) {
-                return responseArr[0];
-              });
-              var _map = {};
-              _.each(_panelMetaArr, function(_panelMeta) {
-                _.each(_panelMeta.genes, function(_gene) {
-                  if (!_map.hasOwnProperty(_gene.hugoGeneSymbol)) {
-                    _map[_gene.hugoGeneSymbol] = {};
-                    _map[_gene.hugoGeneSymbol].panel_id = [];
-                    _map[_gene.hugoGeneSymbol].sample_num = 0;
+
+          if (_panels.length > 0) {
+            $.ajax({
+              type: 'POST',
+              url: window.cbioURL + 'api/gene-panels/fetch?projection=DETAILED',
+              data: JSON.stringify(_panels),
+              dataType: 'json',
+              contentType: "application/json; charset=utf-8",
+              success: function(panels) {
+                _.each(panels, function(panel) {
+                  _.each(panel.genes, function(gene) {
+                    if (!geneSampleMap[gene.hugoGeneSymbol]) {
+                      geneSampleMap[gene.hugoGeneSymbol] = {
+                        sampleUids: []
+                      }
+                    }
+                    geneSampleMap[gene.hugoGeneSymbol].sampleUids = geneSampleMap[gene.hugoGeneSymbol].sampleUids.concat(panelSamplesMap[panel.genePanelId]);
+                  })
+                });
+                if (panelSamplesMap.hasOwnProperty(wholeExomeSequencedPanelId)) {
```

Comment:
may be you could move this condition after line 1422

Reply:
Nope, you don't want to do a O(n2) if O(1) works here.